Add external ID and version fields

mattcookio opened the pull request.

Commit:
Add version and externalID as config options

Commit:
v1.0.0

Commit:
Update sdk dependency

Commit:
Revert "Update sdk dependency"

This reverts commit 24236d68323729211f93adfd2f7cd6112400e165.

Commit:
Update dependency away from beta

Commit:
Merge pull request #1 from adastradev/feature/service-versioning

Feature/service versioning

aregier commented:
Hi Matt! It's been too long since we've had a chance to chat. I'm excited to see activity in these projects for serverless discovery. I can't see the corresponding changes to the service that this hits.
Will you be planning on open-sourcing those enhancements as well? (in https://github.com/adastradev/serverless-discovery-service-aws)

Please let me know if you need anything, including ownership change. I don't want to hold you up.

mattcookio commented:
@aregier I totally agree! Hope you're doing well. We do plan on open sourcing them - looks like the changes were made on Bitbucket and haven't been copied over. Hopefully sometime tomorrow I'll get that repository updated and ping you (I just need to get write access).

We're running on a forked @adastradev/serverless-discovery-plugin npm package (public) just so we could get it running quickly today with the updates, so no worries on that front. Would definitely be open to your thoughts on ownership/sync of these repositories moving forward? I'm not sure how that works on the NPM side of things.

mattcookio commented:
@aregier Should be in sync now. :)

Commit:
Add fallback freeform serviceURL field to serverless.yml

Commit:
Bump version

Commit:
JSON stringify

Commit:
Merge pull request #2 from adastradev/feature/fallback-serviceurl

Add fallback/override serviceURL arg to Serverless.yml

Commit:
Fix lookup call and update sdk

Commit:
Merged in feature/discovery-fix (pull request #1)

Fix lookup call and update sdk

Approved-by: Lindsey Brown
Approved-by: Bo Forrester

snyk-bot committed:
fix: .snyk, package.json & package-lock.json to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/SNYK-JS-AXIOS-174505

awilliams0923 committed:
first set of fixes--does not include the axios fix

awilliams0923 committed:
Merged in snyk-fix-a4803649044f3469c93d279ce185cff7 (pull request #3)

fix: .snyk, package.json & package-lock.json to reduce vulnerabilities

Approved-by: Mike Arps <mdarps@aais.com>
Approved-by: Randy Arms <rarms@aais.com>
Approved-by: Matt Nestler <mnestler@aais.com>
Approved-by: Alysha Williams <awilliams@aais.com>
Approved-by: Chris Alt <calt@aais.com>

mdarpsaais committed:
swap service URL parameter checking to use
user defined serviceURL first if present

mdarpsaais committed:
add feature to read and store
deployment properties

mdarpsaais committed:
bump version

mdarpsaais committed:
Merged in feature/serviceURLfix (pull request #5)

Feature/serviceURLfix - add feature to read and store deployment properties

Approved-by: Cory Routledge
Approved-by: Randy Arms

aneihouse committed:
Move snyk to dev dependencies

aneihouse committed:
Merged in astra-12230 (pull request #6)

Move snyk to dev dependencies

Approved-by: Brian Wetzel <bwetzel@aais.com>
Approved-by: Aaron Neihouse <aneihouse@aais.com>

TwoSlick committed:
Dependency updates

TwoSlick committed:
Removed a ts-ignore

TwoSlick committed:
Reverted a file

TwoSlick committed:
Fixed release file includes

TwoSlick committed:
Adding the Bitbucke pipeline config

TwoSlick committed:
Merged in feature/depupdate (pull request #7)

Dependency updates

Approved-by: Lindsey Hermanson
Approved-by: Cory Routledge

croutledge committed:
Added codecov

croutledgeaais committed:
Merged in feature/codecov (pull request #9)

Added codecov

Approved-by: Joel Wiebe
Approved-by: Lindsey Hermanson
Approved-by: Tim Rupe
Approved-by: Matt Campbell
